core/src/main/scala/kafka/log/ProducerStateManager.scala

```diff
@@ -138,49 +147,58 @@ private[log] class ProducerIdEntry(val producerId: Long, val batchMetadata: muta
  *                      the most recent appends made by the producer. Validation of the first incoming append will
  *                      be made against the lastest append in the current entry. New appends will replace older appends
  *                      in the current entry so that the space overhead is constant.
- * @param validateSequenceNumbers Whether or not sequence numbers should be validated. The only current use
- *                                of this is the consumer offsets topic which uses producer ids from incoming
- *                                TxnOffsetCommit, but has no sequence number to validate and does not depend
- *                                on the deduplication which sequence numbers provide.
- * @param loadingFromLog This parameter indicates whether the new append is being loaded directly from the log.
- *                       This is used to repopulate producer state when the broker is initialized. The only
- *                       difference in behavior is that we do not validate the sequence number of the first append
- *                       since we may have lost previous sequence numbers when segments were removed due to log
- *                       retention enforcement.
+ * @param validationType Indicates the extent of validation to perform on the appends on this instance. Offset commits
+ *                       coming from the producer should have EpochOnlyValidation. Appends which aren't from a client
+ *                       will not be validated at all, and should be set to NoValidation. All other appends should
+ *                       have FullValidation.
```

Comment:
Seems like we didn't update the enum names in this comment. My suggestion is to refer to `ValidationType` values for the explanation and include some javadoc for each type. This comment includes the information for `EpochOnly`, we should ideally include some for `None` an `Full`. All the important information is available in the PR, so it's just a matter of moving it to a place that people will see when trying to understand why we don't validate in some cases. Maybe that can be done together with your NPE fix @apurvam. Thanks!

Reply:
Updated the documentation here: https://github.com/apache/kafka/pull/4022/commits/814434f4f91f40f6db9b3e6da08bb61d90265911